engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/extractor/DataSourceSqlExtractor.scala

```diff
@@ -0,0 +1,82 @@
+package pl.touk.nussknacker.engine.flink.table.extractor
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.flink.table.api.{EnvironmentSettings, TableEnvironment}
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+import pl.touk.nussknacker.engine.flink.table.extractor.SqlStatementNotExecutedError.statementNotExecutedErrorDescription
+import pl.touk.nussknacker.engine.flink.table.extractor.SqlStatementReader.SqlStatement
+import pl.touk.nussknacker.engine.flink.table.extractor.TypeExtractor.extractTypingResult
+
+import scala.jdk.OptionConverters.RichOptional
+import scala.util.{Failure, Success, Try}
+
+object DataSourceSqlExtractor extends LazyLogging {
+
+  import scala.jdk.CollectionConverters._
+
+  // TODO: Make this extractor more memory/cpu efficient and ensure closing of resources
+  def extractTablesFromFlinkRuntime(
+      sqlStatements: List[SqlStatement]
+  ): DataSourceSqlExtractorResult = {
+    val settings = EnvironmentSettings
+      .newInstance()
+      .build()
+    val tableEnv = TableEnvironment.create(settings)
+
+    val sqlErrors = sqlStatements.flatMap(s =>
+      Try(tableEnv.executeSql(s)) match {
+        case Failure(exception) => Some(SqlStatementNotExecutedError(s, exception))
+        case Success(_)         => None
+      }
+    )
+
+    val tableDefinitions = for {
+      catalogName  <- tableEnv.listCatalogs().toList
+      catalog      <- tableEnv.getCatalog(catalogName).toScala.toList
+      databaseName <- catalog.listDatabases.asScala.toList
+      tableName    <- tableEnv.listTables(catalogName, databaseName).toList
+      // table path may be different for some catalog-managed tables - for example JDBC catalog adds a schema name to
+      // table name when listing tables, but querying using tablePath with catalog.database.schema.tableName throws
+      // exception
+      tablePath = s"$catalogName.$databaseName.$tableName"
```

Comment:
How flink-sql deal with such a long names? Haven't they got some kind of default catalog / default database concepts?

Reply:
Normally you can use `USE` statements which change env context - https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/dev/table/sql/use/#use-catalog. There is also the default catalog and database created when table env is initialized. In this implementation I wanted to not depend on context.